src/Graphics/Effect/Effect.cs

```diff
@@ -940,41 +953,71 @@ private unsafe void INTERNAL_parseEffectStruct(IntPtr effectData)
 							{
 								memSize *= mem[j].info.elements;
 							}
-							memList.Add(new EffectParameter(
+							string cachedString = String.Empty;
+							if (mem[j].info.parameter_type == MOJOSHADER_symbolType.MOJOSHADER_SYMTYPE_STRING)
+							{
+								MOJOSHADER_effectObject* objectsPtr = (MOJOSHADER_effectObject*)effectPtr->objects;
+								int* index = param.value.values + curOffset;
+								cachedString = Marshal.PtrToStringAnsi(objectsPtr[*index].stringvalue.stringvalue);
+							}
+
+
+							EffectParameter toAdd = new EffectParameter(
 								Marshal.PtrToStringAnsi(mem[j].name),
 								null,
-								(int) mem[j].info.rows,
-								(int) mem[j].info.columns,
-								(int) mem[j].info.elements,
-								XNAClass[(int) mem[j].info.parameter_class],
-								XNAType[(int) mem[j].info.parameter_type],
+								(int)mem[j].info.rows,
+								(int)mem[j].info.columns,
+								(int)mem[j].info.elements,
+								XNAClass[(int)mem[j].info.parameter_class],
+								XNAType[(int)mem[j].info.parameter_type],
```

Comment:
I bet Visual Studio did this one...

```suggestion
								(int) mem[j].info.rows,
								(int) mem[j].info.columns,
								(int) mem[j].info.elements,
								XNAClass[(int) mem[j].info.parameter_class],
								XNAType[(int) mem[j].info.parameter_type],
```

Reply:
That was MonoDevelop

Reply:
Aw, man... guess I forget what MonoDevelop can do when formatting is turned on (I have it pretty much all the way off now).